src/controllers/management-controllers/asset-controller.ts

```diff
@@ -11,23 +11,27 @@ import { Inner } from "../../inner";
 
 export class AssetController {
   #inner: Inner;
+  #context?: EdcConnectorClientContext;
   defaultContextValues = {
     edc: EDC_CONTEXT,
   };
 
-  constructor(inner: Inner) {
+  constructor(inner: Inner, context?: EdcConnectorClientContext) {
     this.#inner = inner;
+    this.#context = context;
   }
 
   async create(
-    context: EdcConnectorClientContext,
     input: AssetInput,
+    context?: EdcConnectorClientContext,
```

Comment:
I don't think this is possible in TS, but I'll be happy to be denied.

Breaking changes already appeared when we separated the controllers, we still are in an incubating phase, it won't hurt that much to have such changes now.

Reply:
You should be able to overload methods by adding a declaration before the method's signature:

```ts
async create(context: EdcConnectorClientContext, input: AssetInput): Promise<CreateResult>;
async create(input: AssetInput): Promise<CreateResult>;
async create(context: EdcConnectorClientContext | AssetInput, input?: AssetInput): Promise<CreateResult> {
  // ...
}
```

Reply:
I tried like this:
```ts
async create(context: EdcConnectorClientContext, input: AssetInput): Promise<IdResponse>;
async create(input: AssetInput): Promise<IdResponse>;
async create(context?: EdcConnectorClientContext | AssetInput, input?: AssetInput): Promise<IdResponse>
```

but it gives me error because in the third `context` can be both context and input so it's not able to figure out the correct `context` properties...

I tried other combinations but it's not working... 

I think a breaking change at this point is more bearable than a confusing method signature. If needed I would create a complete different class and deprecate the current one, but, it feels a little overkill.

Reply:
Uhm, ok fair - let's keep the current order. Thank you for keeping up with me, @ndr-brt